packages/vue-components/src/PageNavButton.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <div v-show="hasPageNav">
+    <span
+      id="toggle-page-nav-button"
+      :class="['glyphicon', { 'nav-menu-close-icon': show }]"
+      @click="togglePageNav"
+    ></span>
+    <div ref="pageNavMenu" :class="['page-nav-menu', { 'nav-menu-open': show }]"></div>
+  </div>
+</template>
+
+<script>
+import $ from './utils/NodeList';
+
+export default {
+  data() {
+    return {
+      hasPageNav: false,
+      show: false,
+    };
+  },
+  inject: ['initPageNav'],
+  methods: {
+    togglePageNav() {
+      this.show = !this.show;
+      this.toggleSiteNavButton();
+    },
+    toggleSiteNavButton() {
+      const siteNavButton = document.getElementById('toggle-site-nav-button');
+      if (siteNavButton !== null) {
+        if (this.show) {
+          siteNavButton.classList.add('hide-nav-button');
+        } else {
+          siteNavButton.classList.remove('hide-nav-button');
+        }
+      }
+    },
+  },
+  mounted() {
+    const buildPageNavMenu = (pageNavItems) => {
+      this.initPageNav();
+      this.hasPageNav = true;
+      const { pageNavMenu } = this.$refs;
+      for (let i = 0; i < pageNavItems.childNodes.length; i += 1) {
+        pageNavMenu.appendChild(pageNavItems.childNodes[i].cloneNode(true));
+      }
+    };
+
+    const $el = $(this.$refs.pageNavMenu);
+    const pageNav = document.getElementById('page-nav');
```

Comment:
cloning the html directly would disable reactivity / other things if the `site-nav` / `page-nav` has reactive elements that have already been processed in some way (e.g. vue components)

you can refer to `retriever.vue` (used in `<panel src="...">`) for mounting a clone of the original html source (as received by the browser).
However, it makes a webrequest (although it would likely hit the cache), so if there's a 'native' method to retrieve the original page source, all the more better. If not, directly reusing `<retriever>` should be fine too.

Reply:
Alright I've decided to use `<retriever>` to get the contents of the identifier `id=page/site-nav`. However, if these identifiers are not present, will have to look for their respective classes still so it will fallback to using `appendChild`. Not sure if that's fine?